### ISSUE TITLE
undeclared identifiers inside with statement 

### ISSUE THREAD

**Issue by Sorien** [casteng/i-pascal#28]
```pascal
unit Unit1;

interface

type
  TFoo = class
  public
    constructor Create; overload;
    procedure Fa();
  end;

procedure Test();

implementation

procedure Test();
begin
  with TFoo.Create do
  try
    Fa; //undeclared identifier
  finally
    Free; //undeclared identifier
  end;
end;


{ TFoo }

constructor TFoo.Create;
begin

end;

procedure TFoo.Fa;
begin

end;

end.
```

**Comment by casteng** [comment]
There is a problem with type inference of a constructor call. Will be fixed in next release.

**Comment by casteng** [comment]
Fixed in 1.90
